ingestion/src/metadata/ingestion/source/database/domodatabase.py

```diff
@@ -0,0 +1,199 @@
+        self.config = config
+        self.source_config: DatabaseServiceMetadataPipeline = (
+            self.config.sourceConfig.config
+        )
+        self.metadata = OpenMetadata(metadata_config)
+        self.service_connection = self.config.serviceConnection.__root__.config
+        self.status = SQLSourceStatus()
+        self.connection = get_connection(self.service_connection)
+        self.domo_client = self.connection.client
+        self.client = DomoClient(self.service_connection)
+        super().__init__()
+
+    @classmethod
+    def create(cls, config_dict: dict, metadata_config: OpenMetadataConnection):
+        config = WorkflowSource.parse_obj(config_dict)
+        connection: DomoDatabaseConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, DomoDatabaseConnection):
+            raise InvalidSourceException(
+                f"Expected DomoDatabaseConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_database_names(self) -> Iterable[str]:
+        database_name = "default"
+        yield database_name
+
+    def yield_database(self, database_name: str) -> Iterable[CreateDatabaseRequest]:
+        yield CreateDatabaseRequest(
+            name=database_name,
+            service=EntityReference(
+                id=self.context.database_service.id,
+                type="databaseService",
+            ),
+        )
+
+    def get_database_schema_names(self) -> Iterable[str]:
+        scheme_name = "default"
+        yield scheme_name
+
+    def yield_database_schema(
+        self, schema_name: str
+    ) -> Iterable[CreateDatabaseSchemaRequest]:
+        yield CreateDatabaseSchemaRequest(
+            name=schema_name,
+            database=EntityReference(id=self.context.database.id, type="database"),
+        )
+
+    def get_tables_name_and_type(self) -> Optional[Iterable[Tuple[str, str]]]:
+        schema_name = self.context.database_schema.name.__root__
+        table_name = ""
+        try:
+            tables = list(self.domo_client.datasets.list())
+            for table in tables:
+                table_name = table["id"]
+                table_name = self.standardize_table_name(schema_name, table_name)
+                table_fqn = fqn.build(
+                    self.metadata,
+                    entity_type=Table,
+                    service_name=self.context.database_service.name.__root__,
+                    database_name=self.context.database.name.__root__,
+                    schema_name=self.context.database_schema.name.__root__,
+                    table_name=table_name,
+                )
+
+                if filter_by_table(
+                    self.config.sourceConfig.config.tableFilterPattern,
+                    table_fqn
+                    if self.config.sourceConfig.config.useFqnForFiltering
+                    else table_name,
+                ):
```

Comment:
also here while filtering we are basically filtering on table id instead of the table name, let's fix that and test if the table filter pattern works properly